src/SimpleVersion.Core/Environment/TeamCityEnvironment.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed under the MIT license. See https://kieranties.mit-license.org/ for full license information.
+
+using System.Text.RegularExpressions;
+using SimpleVersion.Extensions;
+
+namespace SimpleVersion.Environment
+{
+    /// <summary>
+    /// Represents the environment when running on Azure Devops.
+    /// </summary>
+    public class TeamCityEnvironment : BaseVersionEnvironment
+    {
+        private static readonly Regex _trim = new Regex(@"^refs\/(heads\/)?", RegexOptions.IgnoreCase);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TeamCityEnvironment"/> class.
+        /// </summary>
+        /// <param name="accessor">The accessor for environment variables.</param>
+        public TeamCityEnvironment(IEnvironmentVariableAccessor accessor)
+            : base(accessor)
+        {
+            CanonicalBranchName = Variables.GetVariable("BUILD_SOURCEBRANCH");
```

Comment:
Does TeamCity actually use this variable?  I cannon find any resource referring to this.  I can find https://www.jetbrains.com/help/teamcity/predefined-build-parameters.html#VCS+Parameters which involves the vcsroot and also references to `teamcity.build.branch` - it seems it should be referencing one of those?

Failing that, we may need to introduce a custom environment variable that you will need to explicitly configure ahead of running  Simpleversion.  Similar to GitVersion.

Unfortunately I doe not have a TeamCity environment for testing: https://gitversion.net/docs/reference/build-servers/teamcity

Reply:
I couldn't find any predefined environment variables that Teamcity would use out of the box. anything without the env. prefix is not easily accessible during the build either. 
Since there's nothing specific "prepared option" i just went with the BUILD_SOURCEBRANCH that you were using in the TF environment. I'll happily change it to other names. 
And yes, i fully expect having to setup an env.BUILD_SOURCEBRANCH and point it at the teamcity.* branch variables of the used vcs root - did the same for gitversion back in the day. 
For internal testing i'm setting both the env.BUILD_SOURCEBRANCH and the env.TF_BUILD=true and that works fine, i'd just like to avoid setting TF_BUILD, hence the new environment :)

Reply:
Good stuff - on reflection of the issue then we may actually be able to simplify this a lot.
We can specify a common fallback environment variable `simpleversion.sourcebranch` which if provided will act as the fallback for _any_ build system.
Effectively we can say if that environment variable is set, it wins.

So you set it in TC and you're good.  For any other build system, we use the system specific environment, but override with `simpleversion.sourcebranch` if it is set.

I will take a look.

Reply:
@anmo13 I've introduced a clean fallback and documentation in #146 - Please check out and take a look - I think it should solve your issues and will make it simpler for others in the long run.
This: https://github.com/Kieranties/SimpleVersion/pull/146/files#diff-5cb66f28e120a6052401a3ba4601a64a6fde0068e1fa7f3107fbe0eec5ef335c describes how to configure things

Reply:
@Kieranties That looks great, thanks - i'll configure that then once its in. 
Would also appreciate a non-beta version once you're ready for that.

Reply:
Apologies - for the delay - this will go into a beta today.

Reply:
@Kieranties no worries. Will give it a spin the moment i find it on nuget.org :)

Reply:
This is now live